### DISCUSSION TITLE
Battery issue in TM1701

### DISCUSSION THREAD

**Discussion by wildanbaraja** [daliansky/XiaoMi-Pro#624] [Q&A]
Hello, greetings from Indonesia. Thank you for providing this setting. I'm glad I came to this page. 

I currently have a problem, I have a 15inch mi notebook intel core i7 8550u, when I use bigsur I have battery problems. Battery life was fine at first, but after a month of using it suddenly reduced battery life. And one time I couldn't boot into mac os, I turned on my laptop, and suddenly it shuts down which shows that the battery is empty, sometimes when on the desktop, sometimes when on the mi logo. And after some time, this problem also occurs when I charge my laptop. I think my battery needs to be replaced. But I tried to remove mac os and install linux and windows, the battery life is good, it can last 2 hours. Do you think maybe I misconfigured opencore? Or maybe my battery needs to be patched?

Thank you in advance.

**Comment by KimRay1** [comment]
The same issue was appeared in my laptop.My battery was damaged and actual capacity of the battery is reduced by 5% when i installed BigSur for my laptop.Now,I haven't found a similar problem yet since I installed Monterey for my laptop.Maybe it's the poor battery management strategy of BigSur that's causing the problem.Anyway,you can try to install Monterey for your laptop without worrying about your laptop's battery.By the way,my laptop's model is xiaomi notebook Pro GTX with i5 8250U and GTX 1050 MAX-q.

**Comment by stevezz3** [comment]
Our battery works out of box by using [SMCBatteryManager](https://github.com/acidanthera/VirtualSMC) from VirtualSMC or [ACPIBatteryManager](https://bitbucket.org/RehabMan/os-x-acpi-battery-driver/downloads/). However, since our Xiaomi Pro laptop does not have EmbeddedControl Regions, I do not know the detail of how our battery registers work. On EC supported laptops, we can implement supplement battery information by doing https://github.com/acidanthera/VirtualSMC/blob/master/Docs/Battery%20Information%20Supplement.md. But on our laptop, the battery information is a mystery hidden somewhere in DSDT.

The other thing that stops me to dig deeper is confusing variable names in _BIF method. Our laptop ACPI developer leaves meaningless variable names for battery registers that are not easy to understand.

**Comment by MarFre22** [comment]
In my case, after installing Monterey I experienced better battery duration compared to Catalina (I didn't try BigSur). But if I leave any USB connected after closing the lid, It can't hibernate/suspend, my battery empties. 
Possible solution: disconnect any USB dongles before suspend.

**Comment by wildanbaraja** [comment]
it turns out that my battery is the problem. I have this problem on every OS. Especially when my battery ran out. 
My temporary solution is i need to plug out the battery socket and plug it back, and it's good to go for a couple hours after recharging.
